### DISCUSSION TITLE
Unable to use ForJob() to associate triggers

### DISCUSSION THREAD

**Discussion by y1meng** [quartznet/quartznet#1945] [Q&A] [answered]
`  
public static ITrigger TestForJob() {
            var job1CronString = scheduleConfig["job1"];

            var job = JobBuilder.Create<HelloJob>()
                .WithIdentity("Job","JobGroup")
                .Build();

            var trigger = TriggerBuilder.Create()
                .WithIdentity("JobTrigger", "TriggerGroup")
                .WithCronSchedule(job1CronString)
                .ForJob(job)
                .Build();

            return trigger;
 }
//run job
    private static async void RunScheduleJob(IServiceProvider serviceProvider) {

            var scheduler = serviceProvider.GetRequiredService<ISchedulerFactory>().GetScheduler().Result;

            var trigger = SchedulerJobConfig.TestForJob();
            await scheduler.ScheduleJob(trigger);
}

`
This error message was returned, I don't understand what I'm doing wrong, any help is appreciated ,**Quartz.JobPersistenceException:“The job (JobGroup.Job) referenced by the trigger does not exist**.”

**Comment by lahma** [comment, marked as answer]
You need to register both job and trigger. The documentation is a good place to start:

https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/overview.html

**Reply by y1meng** [reply]
> You need to register both job and trigger. The documentation is a good place to start:
> 
> https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/overview.html

It is no problem to register triggers and jobs at the same time,But what should I do to use this overloaded method?
![image](https://user-images.githubusercontent.com/45099938/224309813-759678c9-3f4c-4b34-a4df-0b77d3132f9f.png)

**Reply by lahma** [reply]
You need to use AddJob before that call. Did you read the documentation?

**Reply by y1meng** [reply]
> You need to use AddJob before that call. Did you read the documentation?

I'm very sorry, my English is not good and I may have missed some details,But I followed the best practice at first, I added static Key, it didn't work, then I tried to use ForJob to associate (my own attempt), sorry again, I may have missed some documentation content
https://www.quartz-scheduler.net/documentation/best-practices.html#job-tips

**Reply by lahma** [reply]
Use AddJob before using ScheduleJob.

**Reply by y1meng** [reply]
> Use AddJob before using ScheduleJob.

ok thanks for your help
